K8s-Orch-Tools/trigger_jenkins.py

```diff
@@ -0,0 +1,257 @@
+#  username and password are identical for all nodes.  You specify the password
+#  via command line as -p <password>, or throught the environment variable
+#  NODE_PASSWORD.
+#
+#  By default the username is "root", but this can be overridden via the command
+#  line as "--node-user <username>".
+#
+#
+# Script Output
+# =============
+# After the Job has been run, the artifacts created by the job, the
+# console output, and a summary suitable for pasting into a report
+# are created in ./setup-cortx-cluster-solution-input_<build-number>.
+#
+#
+########################################################
+
+
+
+import argparse
+import datetime
+import re
+import os
+import sys
+import time
+
+from yaml import load, Loader
+import requests
+
+import urllib3
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+JENKINS_JOB = "https://eos-jenkins.colo.seagate.com/job/Cortx-Kubernetes/job/setup-cortx-cluster-solution-input"
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-s', '--solution', dest='solution_file', required=True)
+parser.add_argument('-t', '--tag', dest='cortx_k8s_tag', required=True)
+parser.add_argument('--node-user', dest='node_user', default='root')
+parser.add_argument('-p', '--node-password', dest='node_password')
+parser.add_argument('-u', '--user-token', dest='user_token')
+#Note: For --debug-skip-trigger, the arg shuld be the build url
+#      For example:  "http://eos-jenkins.colo.seagate.com/job/Cortx-kubernetes/job/setup-cortx-cluster-solution-input/341/"
+parser.add_argument('--debug-skip-trigger', dest='debug_skip_trigger')
+
+args = parser.parse_args()
+
+if not args.user_token:
+    args.user_token = os.environ.get('JENKINS_USER_TOKEN')
+    if not args.user_token:
+        print("No Jenkins user token specified.", file=sys.stderr)
+        print('  Specify with -u/--user-token or with', file=sys.stderr)
+        print('  the environment variable "JENKINS_USER_TOKEN"', file=sys.stderr)
+        sys.exit(1)
+
+if not re.match('\w+:\w+', args.user_token):
+    print(f"Invalid Jenkins user token specified ({args.user_token}).", file=sys.stderr)
+    print("The user token should have the form \"<jenkins-username>:<jenkins-token>\".", file=sys.stderr)
+    sys.exit(1)
+
+username, password = args.user_token.split(':')
+
+if not args.node_password:
+    args.node_password = os.environ.get('NODE_PASSWORD')
+    if not args.node_password:
+        print('No password for cortx nodes specified.', file=sys.stderr)
+        print('  Specify with -p/--node-password or with', file=sys.stderr)
+        print('  the environment variable "NODE_PASSWORD"', file=sys.stderr)
+        sys.exit(1)
+
+#
+# Read hostnames from solution.yaml
+#
+if not os.path.exists(args.solution_file):
+    print("File {args.solution_file} does not exist.", file=sys.stderr)
+    sys.exit(1)
+
+solution_content = load(open(args.solution_file), Loader=Loader)
```

Comment:
```suggestion
solution_content = yaml.safe_load(open(args.solution_file))
```
Is it necessary to use `load()`? I think the solution file is fine with `safe_load()`, then you also don't need `Loader`.